lib/include/openamp/virtio.h

```diff
@@ -277,6 +283,54 @@ struct virtio_dispatch {
 	void (*notify)(struct virtqueue *vq);
 };
 
+/**
+ * @brief Alloc a buffer from virtio transport layer.
+ *
+ * @param vdev		Pointer to virtio device structure.
+ * @param size		Alloc buffer size.
+ * @param align		Buffer alignment.
+ *
+ * @return Pointer to the return buffer. NULL indicates no enough buffer
+ *         or some errors happened.
+ */
+static inline void *virtio_alloc_buf(struct virtio_device *vdev,
+				     size_t size, size_t align)
+{
+	return vdev->func->alloc_buf(vdev, size, align);
+}
+
+/**
+ * @brief Alloc a buffer from virtio transport layer with zero value.
+ *
+ * @param vdev		Pointer to virtio device structure.
+ * @param size		Alloc buffer size.
+ * @param align		Buffer alignment.
+ *
+ * @return Pointer to the return buffer. NULL indicates no enough buffer
+ *         or some errors happened.
+ */
+static inline void *virtio_zalloc_buf(struct virtio_device *vdev,
+				      size_t size, size_t align)
+{
+	void *ptr = virtio_alloc_buf(vdev, size, align);
+
+	if (ptr)
+		memset(ptr, 0, size);
+
+	return ptr;
+}
+
+/**
+ * @brief Free a buffer alloced from transport layer.
+ *
+ * @param vdev		Pointer to virtio device structure.
+ * @param buf		Pointer to an allocated buffer from transport layer.
+ */
+static inline void virtio_free_buf(struct virtio_device *vdev, void *buf)
+{
+	vdev->func->free_buf(vdev, buf);
```

Comment:
Check argumentts:

* vdev -> return -EINVAL if NULL;
* free_buf -> function may be not supported, check if it exists first and return -ENXIO if NULL

Reply:
Done